src/modules/product-category/product-category.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable, InternalServerErrorException} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { ProductCategory } from './entities/product-category.entity';
+import { CreateProductCategoryDto } from './dto/create-product-category.dto';
+
+@Injectable()
+export class ProductCategoryService {
+  constructor(
+    @InjectRepository(ProductCategory)
+    private categoryRepository: Repository<ProductCategory>
+  ) {}
+
+  async createProductCategory(CreateProductCategoryDto: CreateProductCategoryDto): Promise<ProductCategory> {
+   try {
+    const category = this.categoryRepository.create(CreateProductCategoryDto);
+    return await this.categoryRepository.save(category);
+   } catch (error) {
```

Comment:
```suggestion
 const category = this.categoryRepository.create(CreateProductCategoryDto);
 await this.categoryRepository.save(category);
 
 return {
     // The product category
 }
```

Reply:
do you mean that i should assign a variable to  await this.categoryRepository.save(category); 
and then return it